### DISCUSSION TITLE
Unreal Engine 5 and GitHub - Issues with Landscape

### DISCUSSION THREAD

**Discussion by AddamarDan** [community/community#109239] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Body

Hello,

I have recently started a project with a couple of friends, and we are using GitHub as our go-to-software to get assets, progress from eachother, etc.

There is one problem though, if me and my friends are editing the same Landscape (either sculpt it, smoothen it, paint it, basically everything that is altering the landscape shape/color), we will have conflicts when we commit and the changes are not merging but instead, making the edited landscape (by both sides) disappear completely when we update from “main branch”.

Can anyone help us figure this out? It’s ruining our project and we are stuck with this issue.

Thank you very much for your help in advance!

Here is a pic how the project looks after we both edited the landscape (me from my end, and my friend from the other, then commiting on github and updating from the “main branch”, instead of the landscape merging together, it completely disappeared from both ends)

![2024-02-19 21_15_17-Window](https://github.com/community/community/assets/145563987/1177fc39-9262-46b1-92a3-2de20aa141e0)

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @AddamarDan ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Comment by neilharrison01** [comment]
This is a common issue when using Git with Unreal Engine. Landscape assets are stored as binary .uasset files, so Git can't merge changes made by two people to the same landscape. Instead, it treats them as conflicting versions, and one change usually overwrites the other.

A few approaches that can help are:

Avoid having multiple people edit the same landscape at the same time.
Split the world into smaller sections using World Partition or separate streaming levels so each person works on different cells.
Use Git LFS for large Unreal assets to manage binary files more reliably, although it won't solve merge conflicts.
If your team frequently collaborates on the same map, you may want to look into Unreal's Multi-User Editing system, which is designed for real-time collaborative editing.

Git works very well for source code, but binary assets like landscapes, materials, and Blueprints require a workflow that minimizes simultaneous edits.
